src/modules/tools/zprobe/CartGridStrategy.cpp

```diff
@@ -396,11 +470,32 @@ float CartGridStrategy::findBed()
 bool CartGridStrategy::doProbe(Gcode *gc)
 {
     gc->stream->printf("Rectangular Grid Probe...\n");
+
+    if(only_by_two_corners){
+        if(gc->has_letter('X') && gc->has_letter('X') && gc->has_letter('A') && gc->has_letter('B')){
+            x_start = gc->get_value('X'); // override default probe start point, will get saved
+            y_start = gc->get_value('Y'); // override default probe start point, will get saved
+            x_size = gc->get_value('A'); // override default probe width, will get saved
+            y_size = gc->get_value('B'); // override default probe length, will get saved
+        } else {
+            gc->stream->printf("In only_by_two_corners mode all XYAB parameters needed\n");
+            return false;
+        }
+    } else {
+        if(gc->has_letter('X')) x_size = gc->get_value('X'); // override default probe width, will get saved
+        if(gc->has_letter('Y')) y_size = gc->get_value('Y'); // override default probe length, will get saved
+    }
+    
     setAdjustFunction(false);
     reset_bed_level();
 
-    if(gc->has_letter('X')) x_size = gc->get_value('X'); // override default probe width, will get saved
-    if(gc->has_letter('Y')) y_size = gc->get_value('Y'); // override default probe length, will get saved
+    if(gc->has_letter('I')) current_grid_x_size = gc->get_value('I'); // override default grid x size
+    if(gc->has_letter('J')) current_grid_y_size = gc->get_value('J'); // override default grid y size
```

Comment:
Adding comment here (same as previous override) so that this doesn't get missed.